sdk/monitor/monitor-opentelemetry-exporter/src/utils/common.ts

```diff
@@ -254,3 +255,21 @@ export function createResourceMetricEnvelope(
   }
   return;
 }
+
+export function serializeAttribute(attribute: AttributeValue | AnyValue | undefined) {
+  if (attribute != undefined) {
+    if (
+      typeof attribute === "string" ||
+      typeof attribute === "number" ||
+      typeof attribute === "boolean"
+    ) {
+      return String(attribute);
+    }
+    if (attribute instanceof Error) {
+      return `{ stack: '${attribute.stack}', message: '${attribute.message}', name: '${attribute.name}'`;
```

Comment:
This method would presumably not serialise any enumerable attributes of the `Error` other than `stack` and `message`? E.g. to cater for a custom type which extends Error.

Is it worth spreading those props and using `JSON.stringify` to cover other enumerable props?

E.g. like:

```typescript
const { message, stack, ...rest } = attribute
return JSON.stringify({ message, stack, ...rest })
```

Reply:
@cuzzlor I had further discussion about this with my team, and we decide to not do any serialization here, Objects and Error objects are not valid LogRecord attributes according to OpenTelemetry [spec](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-attributes), the issue you are experiencing should be fixed upstream in Winston Instrumentation that append the attributes.

Reply:
Update the code, all error properties should be added now